### PR TITLE
fix parsing for common channels, fix log.debug(e) error

### PR DESCRIPTION
**Pull request by JudahSchwartz**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-36600)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 1]
```diff
@@ -2844,12 +2844,14 @@ def init_globals(command_name: str = ''):
     FILE_MIRRORING_ENABLED = demisto.params().get('enable_outbound_file_mirroring', False)
     LONG_RUNNING_ENABLED = demisto.params().get('longRunning', True)
     DEMISTO_API_KEY = demisto.params().get('demisto_api_key', {}).get('password', '')
+    demisto.debug('before demistourls')
     demisto_urls = demisto.demistoUrls()
+    demisto.debug('after demistourls')
```
Comment (JudahSchwartz) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:2849]:
```suggestion
```

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 2]
```diff
@@ -2844,12 +2844,14 @@ def init_globals(command_name: str = ''):
     FILE_MIRRORING_ENABLED = demisto.params().get('enable_outbound_file_mirroring', False)
     LONG_RUNNING_ENABLED = demisto.params().get('longRunning', True)
     DEMISTO_API_KEY = demisto.params().get('demisto_api_key', {}).get('password', '')
+    demisto.debug('before demistourls')
```
Comment (JudahSchwartz) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:2847]:
```suggestion
```

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 3]
```diff
@@ -3003,14 +3022,22 @@ def main() -> None:
     try:
         demisto.info(f'{command_name} started.')
         command_func = commands[command_name]
+        demisto.info('before init globals')
```
Comment (JudahSchwartz) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:3025]:
```suggestion
```

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 4]
```diff
@@ -3003,14 +3022,22 @@ def main() -> None:
...
         init_globals(command_name)
+        demisto.info('after init globals')
         if EXTENSIVE_LOGGING:
             os.environ['PYTHONASYNCIODEBUG'] = "1"
+        demisto.info('before support_multithreading')
```
Comment (JudahSchwartz) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:3030]:
```suggestion
```

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 5]
```diff
@@ -3003,14 +3022,22 @@ def main() -> None:
...
         support_multithreading()
+        demisto.info('after support_multithreading')
```
Comment (JudahSchwartz) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:3032]:
```suggestion
```

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 6]
```diff
@@ -3003,14 +3022,22 @@ def main() -> None:
...
         command_func()  # type: ignore
     except Exception as e:
-        demisto.debug(e)
+        demisto.debug('in exception block')
```
Comment (JudahSchwartz) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:3035]:
```suggestion
```

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 7]
```diff
@@ -3003,14 +3022,22 @@ def main() -> None:
...
+        demisto.error(f'Error occured error: {e}')
+        demisto.error(traceback.format_exc())
+        demisto.debug('debugged it')
```
Comment (JudahSchwartz) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:3038]:
```suggestion
```

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 8]
```diff
@@ -2891,6 +2894,22 @@ def init_globals(command_name: str = ''):
         CACHED_INTEGRATION_CONTEXT = get_integration_context(SYNC_CONTEXT)
 
 
+def parse_common_channels(common_channels: str):
+    common_channels = (common_channels or '').strip()
```
Comment (ShahafBenYakir) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:2896]:
```suggestion
def parse_common_channels(common_channels: str = ''):
    common_channels = (common_channels).strip()
```

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py [hunk 9]
```diff
@@ -5079,6 +5080,29 @@ def test_conversation_history(mocker):
     assert demisto.results.call_args[0][0]['ContentsFormat'] == 'json'
 
 
+@pytest.mark.parametrize('raw, output', [
+    ("""
+    key1:value1
+    key2: value2
```
Comment (ShahafBenYakir) [inline Packs/Slack/Integrations/SlackV3/SlackV3_test.py:5086]: maybe some newlines between the pairs

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 10]
```diff
@@ -2891,6 +2892,22 @@ def init_globals(command_name: str = ''):
...
+    if not common_channels:
+        return {}
+    try:
+        ret = {}
```
Comment (ShahafBenYakir) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:2900]:
```suggestion
        stripped_channels = {}
```

**Review by ShahafBenYakir — APPROVED** [review]

File: Packs/Slack/Integrations/SlackV3/SlackV3.py (hunk 8 above)
Comment (JudahSchwartz) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py:2896]: this doesnt work in the case you got None as the parameter, since it is called explicitly

Comment (ShahafBenYakir) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py:2896]: So isn't it better to change
```
common_channels = demisto.params().get('common_channels', None)
````
to
```
common_channels = demisto.params().get('common_channels', '')
```?
Anyway parse_common_channels is expecting a str, and None is not a str.

**Pull request merged**